autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
 
-    @property
-    def satisfied(self):
-        return self.get_missing(validate=True)
+    def satisfied(self, section=None, compare_checksums=False):
+        """Determines whether all database dependencies are satisfied.
 
-    def prepare_sections(self):
-        """Add database sections to 'databases' if missing.
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        compare_checksums : bool, optional
+            Also check if database information is up-to-date with current hosted databases.
+            (default is False).
 
         Returns
         -------
-        NoneType
+        bool
+            True if all database dependencies are satisfied, otherwise False.
 
         """
-        if not self.config.has_section('databases'):
-            self.config.add_section('databases')
-        for section in Databases.SECTIONS:
-            if self.config.has_option('databases', section):
-                continue
-            outdir = DEFAULT_CONFIG.get('databases', section)
-            self.config.set('databases', section, outdir)
+        any_missing = self.get_missing(section=section, validate=True)
+        if compare_checksums:
+            any_invalid = self.compare_checksums(section=section, validate=True)
+        else:
+            any_invalid = False
+        if not any_missing and not any_invalid:
+            return True
+        else:
+            return False
+
+    def get_remote_checksum(self, section, option):
+        """Get the checksum from provided `section` respective to `option` in `self.config`.
+
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        option : str
+            `option` in `checksums` section corresponding to the section checksum file.
+
+        Returns
+        -------
+        str
+            checksum of remote md5 file. e.g. 'hash filename\n'
+
+        Raises
+        -------
+        ConnectionError
+            Failed to connect to host for provided `option`.
+
+        """
+        if section not in {'ncbi','markers'}:
+            raise ValueError(f'"section" must be "ncbi" or "markers". Provided: {section}')
+        if section == 'ncbi':
+            host = self.config.get(section, 'host')
+            ftp_fullpath = self.config.get('checksums', option)
+            chksum_fpath = ftp_fullpath.split(host)[-1]
+            with FTP(host) as ftp, tempfile.TemporaryFile() as fp:
+                ftp.login()
+                result = ftp.retrbinary(f'RETR {chksum_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{chksum_fpath} download failed')
+                ftp.quit()
+                fp.seek(0)
+                checksum = fp.read().decode()
```

Comment:
Has this been tested? I'm just wondering if it results in the checksum string with a newline at the end, which might cause problems later. Also, if one URL doesn't have a newline, can we assume that all won't?

Reply:
I needed to push before being able to test. I will update with a comment when I have tested the functionality.

Reply:
The checksum check for the markers appears to be working

```python

In [1]: import requests

In [2]: outfpath = 'test.md5'

In [3]: with requests.Session() as session, open(outfpath, 'w') as fh:
   ...:     resp = session.get('https://raw.githubusercontent.com/WiscEvan/Autometa/databases/autometa/databases/markers/bacteria.single_copy.hmm.md5')
   ...:     if not resp.ok:
   ...:         raise 'ouch'
   ...:     fh.write(resp.text)
   ...:

In [4]: resp.text
Out[4]: 'eafdb5bd1447e814a4ee47ee11434ffe bacteria.single_copy.hmm\n'

In [5]: !cat test.md5
eafdb5bd1447e814a4ee47ee11434ffe bacteria.single_copy.hmm
```

Reply:
Both `FTP` and `requests` methods have been tested.